wincode/src/schema/impls.rs

```diff
@@ -1123,3 +1126,162 @@ where
+            zero_copy: true,
+        } => TypeMeta::Static {
+            size,
+            // Note: `&'de T` is NOT zero‑copy in the "raw-bytes representable" sense.
+            // In this crate, `zero_copy: true` means:
+            // - The type's in‑memory representation is exactly its serialized bytes.
+            // - It can be safely initialized by memcpy (no validation, no endianness/layout work).
+            // - Containers may bulk-copy elements (e.g., Vec/BoxedSlice memcpy fast path for Pod).
+            // - It can be deserialized by reference to some underlying source bytes.
+            //
+            // A _reference_ to a zero-copy type does not meet that contract:
+            // - `&'de T` is a pointer with provenance/lifetime, not bytes you can memcpy into place.
+            // - You cannot "initialize a reference" by copying bytes; you must point it at already
+            //   valid storage of `T`.
+            // - Advertising `zero_copy: true` here could incorrectly enable memcpy of reference
+            //   elements (e.g., Vec<&T>), which would be unsound.
+            //
+            // We borrow the underlying `T` here, knowing it is zero-copy, but the reference itself
+            // is never considered zero-copy.
+            zero_copy: false,
+        },
+        // Should be impossible to reach.
+        _ => panic!("Type is not zero-copy"),
+    };
+
+    fn read(reader: &mut impl Reader<'de>, dst: &mut MaybeUninit<Self::Dst>) -> ReadResult<()> {
+        let TypeMeta::Static {
+            size,
+            zero_copy: true,
+        } = T::TYPE_META
+        else {
+            // Should be impossible to reach.
+            unreachable!("Type is not zero-copy");
+        };
+
+        let bytes = reader.borrow_exact(size)?;
+        // SAFETY:
+        // - T::Dst is zero-copy (no invalid bit patterns, no layout requirements, no endianness checks, etc.).
+        // - `bytes.len() == size_of::<T::Dst>()`. `borrow_exact` ensures we read exactly `size` bytes.
+        let val = unsafe { zero_copy::cast_slice_to_t::<T::Dst>(bytes)? };
+        dst.write(val);
+        Ok(())
+    }
+}
+
+impl<'de, T> SchemaRead<'de> for &'de [T]
+where
+    T: SchemaRead<'de> + ZeroCopy,
+{
+    type Dst = &'de [T::Dst];
+
+    const TYPE_META: TypeMeta = match T::TYPE_META {
+        TypeMeta::Static {
+            zero_copy: true, ..
+        } => TypeMeta::Dynamic,
+        _ => panic!("Type is not zero-copy"),
+    };
+
+    fn read(reader: &mut impl Reader<'de>, dst: &mut MaybeUninit<Self::Dst>) -> ReadResult<()> {
+        let TypeMeta::Static {
+            size,
+            zero_copy: true,
+        } = T::TYPE_META
+        else {
+            // Should be impossible to reach.
+            unreachable!("Type is not zero-copy");
+        };
+
+        let len = <BincodeLen>::read::<T::Dst>(reader)?;
+        #[allow(clippy::arithmetic_side_effects)]
+        let bytes = reader.borrow_exact(len * size)?;
```

Comment:
is this overflow case actually worth handling at runtime? deserialized data could be user-provided and even if the overflow actually lowers the size of `bytes` the safety comment below won't hold, i.e. bytes.len() != len * size and 
`unsafe { zero_copy::cast_slice_to_slice_t::<T::Dst>(bytes, len)? }` becomes scary